base_layer/core/src/chain_storage/async_db.rs

```diff
@@ -36,48 +36,75 @@ use crate::{
         types::HashOutput,
     },
 };
+use log::*;
+use rand::{rngs::OsRng, RngCore};
+use std::time::Instant;
 use tari_mmr::MerkleProof;
 
+const LOG_TARGET: &str = "c::bn::async_db";
+
+fn trace_log<F, R>(name: &str, f: F) -> R
+where F: FnOnce() -> R {
+    let start = Instant::now();
+    let trace_id = OsRng.next_u32();
+    debug!(
```

Comment:
Is this not too verbose for debug though?

Reply:
You're right, I've changed it to trace!